pxr/imaging/plugin/hdRpr/mesh.cpp

```diff
@@ -652,7 +654,18 @@ void HdRprMesh::Sync(HdSceneDelegate* sceneDelegate,
             } else {
                 if (m_geomSubsets.size() == m_rprMeshes.size()) {
                     for (int i = 0; i < m_rprMeshes.size(); ++i) {
-                        auto material = getMeshMaterial(m_geomSubsets[i].materialId);
+						bool hasMaterialByShortPath
+							= (sceneDelegate->GetRenderIndex().GetSprim(HdPrimTypeTokens->material, m_geomSubsets[i].materialId)) != nullptr;
+
+						// geomSubset has only relative material path. Relative to mesh.
+						// materials however are stored by full material path
+						// so when relative material path is passed material is not found
+						// thus we have to get full material path to get pointer to material
+						SdfPath parentMesh = this->GetId().GetParentPath();
+						SdfPath relativeMaterialPath = m_geomSubsets[i].materialId.MakeRelativePath(SdfPath::AbsoluteRootPath());
+						SdfPath fullMaterialPath = parentMesh.AppendPath(relativeMaterialPath);
```

Comment:
Not used if `hasMaterialByShortPath==true`. SdfPath manipulations are not free and for some big scenes it can add up to a noticeable amount of time

Reply:
done